pkg/security/security_profile/secprofs.go

```diff
@@ -342,7 +347,7 @@ func (m *Manager) linkProfile(profile *profile.Profile, workload *tags.Workload)
 	profile.Instances = append(profile.Instances, workload)
 
 	// can we apply the profile or is it not ready yet ?
-	if profile.LoadedInKernel.Load() {
+	if profile.LoadedInKernel.Load() && slices.Contains(m.config.RuntimeSecurity.AnomalyDetectionEventTypes, model.SyscallsEventType) {
```

Comment:
Should we add a `isSyscallAnomalyEnabled` boolean field to the Manager struct in order to avoid checking the `AnomalyDetectionEventTypes` slice every time a profile needs to be linked/unlinked?